### PR TITLE
Add component Image360

### PR DESCRIPTION
**Pull request by blingblink**
Add a button, upon clicking will pop up a view for user to explore the 360-degree image of that subworld.

- Add component Image360 at `components\image360\Image360.tsx' : take as input a 360-degree image file,  display the relevant view
- Add component OverlayImage360 at `components\image360\OverlayImage360.tsx': Show the image at an overlay. 
- Add component OverlayImage360Button at `components\image360\OverlayImage360Button.tsx`: Button showing the overlay.
- Redesign Subworld card in `/subworlds/1` and show images from API.
- Add type EnrichedSubworldTemplate and declare Anonymous for empty creator name

## Example

```
<OverlayImage360Button source="https://cdn.discordapp.com/attachments/1096642113074118706/1097885654412697660/kris-guico-rsB-he-ye7w-unsplash.jpg" />
```

### CODE REVIEW

File: components/asset/DerivWorldList.tsx [hunk 1]
```diff
@@ -7,6 +7,8 @@ import { BsFillPeopleFill, BsPlayFill } from "react-icons/bs";
 import { TbWorld } from "react-icons/tb";
 import { TemplateViewModel } from "./RootWorldList";
 import styles from "../../styles/card-item.module.css";
+import OverlayImage360Button from "../image360/OverlayImage360Button";
```
Comment (lth08091998) [inline components/asset/DerivWorldList.tsx:10]: Currently we are doing relative path import, not a good practice (our current technical debt)

Later we can refactor all to absolute path import

File: components/SubworldCard.tsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+                    <OverlayImage360Button
+                        source={data.image}
+                        className="inline-flex w-full justify-center rounded-md bg-white px-3 py-2 text-sm font-semibold text-gray-900 shadow-sm hover:bg-gray-50 sm:col-start-1"
+                  
+                    >
+                        Preview
+                    </OverlayImage360Button>
+                    <button
+                        type="button"
+                        className="inline-flex justify-center rounded-md py-2 px-3 text-sm font-semibold outline-2 outline-offset-2 transition-colors overflow-hidden bg-cyan-500 text-white active:before:bg-transparent hover:before:bg-white/10 active:bg-cyan-600 active:text-white/80 before:transition-colors"
```
Comment (lth08091998) [inline components/SubworldCard.tsx:54]: Curious: how did you come up with these css properties? Did you use some tools to help generate?

File: components/asset/DerivWorldList.tsx [hunk 3]
```diff
@@ -24,10 +26,10 @@ function DerivWorldList(props: ListProps) {
 
     return (
         <section id="nft-list" className="flex flex-col p-2 gap-2 items-center w-[100%]">
-            <div className={`grid grid-cols-1 xl:grid-cols-4 md:grid-cols-2 gap-2`}>
+            <div className={`grid grid-cols-1 xl:grid-cols-3 md:grid-cols-2 gap-2`}>
                 {
                     props.data.map((item, index) =>
-                        <DerivWorldCard key={index} data={item} />)
```
Comment (lth08091998) [inline components/asset/DerivWorldList.tsx:30]: remove this DerivWorldClass if no longer used anywhere else

File: components/SubworldCard.tsx (hunk 2 above)
Comment (blingblink) [reply components/SubworldCard.tsx:54]: They are from a templating website called tailwindui. I reused the components from them with some modifications

File: components/image360/Image360.tsx [hunk 4]
```diff
@@ -0,0 +1,21 @@
+import React, { useEffect, useRef } from 'react';
+import dynamic from 'next/dynamic';
+const ReactPhotoSphereViewer = dynamic(
```
[the remark this answers is not in the document]
Comment (blingblink) [reply components/image360/Image360.tsx:3]: Yes, the image360 library uses “window” from the browser to display. From my understanding, server-side rendering with 360-degree images would be very hard, if not impossible.

**Review by lth08091998 — APPROVED** [review]

**Pull request merged**